docs/commerce/cart/cart.md

```diff
@@ -0,0 +1,100 @@
+they have to log in and transition into an authenticated user.
+
+!!! note 
+
+    For information about roles and permissions that control access to the cart, 
+    see [Permission use cases](../../permissions/permission_use_cases.md#commerce).
+
+### Cart data merging
+
+When a buyer browses the storefront anonymously and fills the cart with items, 
+anonymous cart data is stored in the PHP session storage.
+Then, when an anonymous user logs into the storefront, cart data from the PHP session 
+storage is persisted and merged with any cart information that might already exist in the database for this authenticated user.
+
+If no previous cart data exists, a new cart is created.
+
+### Cart data validation
+
+When a buyer tries to add products to the cart, increase cart item quantity or proceed to checkout, the Cart component performs cart item validation and checks whether:
+
+- the product is available 
+- the requested quantity of product is available 
+- the product is available at a price in the currency selected for the cart 
+
+## Front-end perspective
+
+From the front-end perspective, the cart consists of a main `Cart` object 
+and several widgets.
+`Cart` is a standalone JavaScript object that manages cart data and has no user interface, 
+while widgets consist of JavaScript code and accompanying Twig templates.
+
+### Cart object
+
+The `Cart` object stores cart entry data and a cart summary, which contains additional entry data, like, for example, formatted gross price or validation errors.
+The object exposes several methods, which you can use to get and modify cart entries.
+Only one instance of a `Cart` object can be created.
+
+### Widgets
+
+To customize your store, you can override the Twig templates and extend their logic.
+For more information, see [ Customize storefront layout](../../docs/templating/layout/customize_storefront_layout.md).
+
+#### Main cart
+
+Main cart is the main user interface component of the cart.
+By defaault. it lists items selected for purchase and requested cart item quantities. 
+
+Users who visit the Main cart can remove individual items.
+
+#### Add to cart
+
+By default, the Add to cart component consists of a quantity input field and a button.
+
+#### Minicart
+
+By default, the Minicart component consists of a counter that displays a total number of cart items.
+
+#### Cart summary 
+
+By default, the Cart Summary component displays a subtotal net value of cart lines, 
+a shipping cost disclaimer, a series of tax values applicable to products 
+added in cart, a composition of different taxes, and a total cart value (gross, shipping 
+and taxes included) in the selected currency.
+
+You could extend the Cart summary widget to let buyers navigate from this view, 
+for example, to checkout, or back to shopping, by adding respective buttons.
+
+### Cart service 
+
+The Cart package provides a cart service, which is a module with functions used for calling the [backend API](cart_api.md). 
```

Comment:
The wording doesn't feel right at all. We are talking about PHP class/Symfony service here, not a "module with functions". Also referencing any class should include fully qualified class name i.e. `Ibexa\Contracts\Cart\CartServiceInterface` instead of just `CartServiceInterface` because in many cases you'll see multiple classes using the same name and it won't be clear which one you should use.

```suggestion
The Cart package provides `Ibexa\Contracts\Cart\CartServiceInterface` Symfony service, which is the entrypoint for calling the [backend API](cart_api.md). 
```